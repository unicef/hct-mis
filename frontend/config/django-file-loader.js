const path = require('path');

const {getOptions, interpolateName} = require('loader-utils');
const validate = require('schema-utils');
const schema = require('./options.json');
const {normalizePath} = require('./utils');

function loader(content) {
  const options = getOptions(this);

  validate(schema, options, {
    name: 'File Loader',
    baseDataPath: 'options',
  });

  const context = options.context || this.rootContext;
  const name = options.name || '[contenthash].[ext]';

  const url = interpolateName(this, name, {
    context,
    content,
    regExp: options.regExp,
  });

  let outputPath = url;

  console.log(content.length,"**********************************************************************************************************************************")
  if (options.outputPath) {
    if (typeof options.outputPath === 'function') {
      outputPath = options.outputPath(url, this.resourcePath, context);
    } else {
      outputPath = path.posix.join(options.outputPath, url);
    }
  }

  let publicPath = `${JSON.stringify(outputPath)}`;

  if (options.publicPath) {
    if (typeof options.publicPath === 'function') {
      publicPath = options.publicPath(url, this.resourcePath, context);
    } else {
      publicPath = `${
        options.publicPath.endsWith('/')
          ? options.publicPath
          : `${options.publicPath}/`
      }${url}`;
    }

    publicPath = JSON.stringify(publicPath);
  }

  if (options.postTransformPublicPath) {
    publicPath = options.postTransformPublicPath(publicPath);
  }

  if (typeof options.emitFile === 'undefined' || options.emitFile) {
    const assetInfo = {};

    if (typeof name === 'string') {
      let normalizedName = name;

      const idx = normalizedName.indexOf('?');

      if (idx >= 0) {
        normalizedName = normalizedName.substr(0, idx);
      }

      const isImmutable = /\[([^:\]]+:)?(hash|contenthash)(:[^\]]+)?]/gi.test(
        normalizedName
      );

      if (isImmutable === true) {
        assetInfo.immutable = true;
      }
    }

    assetInfo.sourceFilename = normalizePath(
      path.relative(this.rootContext, this.resourcePath)
    );

    this.emitFile(outputPath, content, null, assetInfo);
  }

  const esModule =
    typeof options.esModule !== 'undefined' ? options.esModule : true;

  return `${esModule ? 'export default' : 'module.exports ='} addPublicPath(${publicPath});`;
}
module.exports = loader;