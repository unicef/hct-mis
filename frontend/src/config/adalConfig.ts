import { AuthenticationContext, adalFetch, withAdalLogin } from "react-adal";
const LOCAL_HOST = "http://localhost:3000";
const redirectUri = () => {
  return process.env.NODE_ENV === "production"
    ? `${process.env.API_BASE_URL}`
    : LOCAL_HOST;
};
export const adalConfig = {
  tenant: "462ad9ae-d7d9-4206-b874-71b1e079776f",
  clientId: "06e50a84-27be-4910-b2ea-43381976c79a",
  redirectUri: redirectUri(),
  endpoints: {
    api: LOCAL_HOST
  },
  cacheLocation: "localStorage",
  postLogoutRedirectUri: redirectUri()
};
export const adalInstance = new AuthenticationContext(adalConfig);
