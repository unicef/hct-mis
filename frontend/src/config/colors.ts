export const DRAWER_TOP_COLOR='#003C8F';