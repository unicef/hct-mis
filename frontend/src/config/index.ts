export const API_BASE_URL = `http://localhost:8082/api`;
export const GRAPHQL_URL = `${API_BASE_URL}/graphql`;
