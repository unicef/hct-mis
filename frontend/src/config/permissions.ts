export const PERMISSIONS = {
  // RDI
  RDI_VIEW_LIST: 'RDI_VIEW_LIST',
  RDI_VIEW_DETAILS: 'RDI_VIEW_DETAILS',
  RDI_IMPORT_DATA: 'RDI_IMPORT_DATA',
  RDI_RERUN_DEDUPE: 'RDI_RERUN_DEDUPE',
  RDI_MERGE_IMPORT: 'RDI_MERGE_IMPORT',

  // Population
  POPULATION_VIEW_HOUSEHOLDS_LIST: 'POPULATION_VIEW_HOUSEHOLDS_LIST',
  POPULATION_VIEW_HOUSEHOLDS_DETAILS: 'POPULATION_VIEW_HOUSEHOLDS_DETAILS',
  POPULATION_VIEW_INDIVIDUALS_LIST: 'POPULATION_VIEW_INDIVIDUALS_LIST',
  POPULATION_VIEW_INDIVIDUALS_DETAILS: 'POPULATION_VIEW_INDIVIDUALS_DETAILS',

  // User Management
  USER_MANAGEMENT_VIEW_LIST: 'USER_MANAGEMENT_VIEW_LIST',

  // Programme
  PRORGRAMME_VIEW_LIST_AND_DETAILS: 'PRORGRAMME_VIEW_LIST_AND_DETAILS',
  PROGRAMME_VIEW_PAYMENT_RECORD_DETAILS:
    'PROGRAMME_VIEW_PAYMENT_RECORD_DETAILS',
  PROGRAMME_CREATE: 'PROGRAMME_CREATE',
  PROGRAMME_UPDATE: 'PROGRAMME_UPDATE',
  PROGRAMME_REMOVE: 'PROGRAMME_REMOVE',
  PROGRAMME_ACTIVATE: 'PROGRAMME_ACTIVATE',
  PROGRAMME_FINISH: 'PROGRAMME_FINISH',

  // Targeting
  TARGETING_VIEW_LIST: 'TARGETING_VIEW_LIST',
  TARGETING_VIEW_DETAILS: 'TARGETING_VIEW_DETAILS',
  TARGETING_CREATE: 'TARGETING_CREATE',
  TARGETING_UPDATE: 'TARGETING_UPDATE',
  TARGETING_DUPLICATE: 'TARGETING_DUPLICATE',
  TARGETING_REMOVE: 'TARGETING_REMOVE',
  TARGETING_LOCK: 'TARGETING_LOCK',
  TARGETING_UNLOCK: 'TARGETING_UNLOCK',
  TARGETING_SEND: 'TARGETING_SEND',

  // Payment Verification
  PAYMENT_VERIFICATION_VIEW_LIST: 'PAYMENT_VERIFICATION_VIEW_LIST',
  PAYMENT_VERIFICATION_VIEW_DETAILS: 'PAYMENT_VERIFICATION_VIEW_DETAILS',
  PAYMENT_VERIFICATION_CREATE: 'PAYMENT_VERIFICATION_CREATE',
  PAYMENT_VERIFICATION_UPDATE: 'PAYMENT_VERIFICATION_UPDATE',
  PAYMENT_VERIFICATION_ACTIVATE: 'PAYMENT_VERIFICATION_ACTIVATE',
  PAYMENT_VERIFICATION_DISCARD: 'PAYMENT_VERIFICATION_DISCARD',
  PAYMENT_VERIFICATION_FINISH: 'PAYMENT_VERIFICATION_FINISH',
  PAYMENT_VERIFICATION_EXPORT: 'PAYMENT_VERIFICATION_EXPORT',
  PAYMENT_VERIFICATION_IMPORT: 'PAYMENT_VERIFICATION_IMPORT',
  PAYMENT_VERIFICATION_VERIFY: 'PAYMENT_VERIFICATION_VERIFY',
  PAYMENT_VERIFICATION_VIEW_PAYMENT_RECORD_DETAILS:
    'PAYMENT_VERIFICATION_VIEW_PAYMENT_RECORD_DETAILS',

  // Grievances
  GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE:
    'GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE',
  GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_LIST_EXCLUDING_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_LIST_SENSITIVE: 'GRIEVANCES_VIEW_LIST_SENSITIVE',
  GRIEVANCES_VIEW_LIST_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_LIST_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_LIST_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_LIST_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE:
    'GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE',
  GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_DETAILS_EXCLUDING_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_DETAILS_SENSITIVE: 'GRIEVANCES_VIEW_DETAILS_SENSITIVE',
  GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_CREATOR:
    'GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_CREATOR',
  GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_OWNER:
    'GRIEVANCES_VIEW_DETAILS_SENSITIVE_AS_OWNER',
  GRIEVANCES_VIEW_HOUSEHOLD_DETAILS: 'GRIEVANCES_VIEW_HOUSEHOLD_DETAILS',
  GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_CREATOR:
    'GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_CREATOR',
  GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_OWNER:
    'GRIEVANCES_VIEW_HOUSEHOLD_DETAILS_AS_OWNER',
  GRIEVANCES_VIEW_INDIVIDUALS_DETAILS: 'GRIEVANCES_VIEW_INDIVIDUALS_DETAILS',
  GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_CREATOR:
    'GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_CREATOR',
  GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_OWNER:
    'GRIEVANCES_VIEW_INDIVIDUALS_DETAILS_AS_OWNER',
  GRIEVANCES_CREATE: 'GRIEVANCES_CREATE',
  GRIEVANCES_UPDATE: 'GRIEVANCES_UPDATE',
  GRIEVANCES_UPDATE_AS_CREATOR: 'GRIEVANCES_UPDATE_AS_CREATOR',
  GRIEVANCES_UPDATE_AS_OWNER: 'GRIEVANCES_UPDATE_AS_OWNER',
  GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE:
    'GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE',
  GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_CREATOR:
    'GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_CREATOR',
  GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_OWNER:
    'GRIEVANCES_UPDATE_REQUESTED_DATA_CHANGE_AS_OWNER',
  GRIEVANCES_ADD_NOTE: 'GRIEVANCES_ADD_NOTE',
  GRIEVANCES_ADD_NOTE_AS_CREATOR: 'GRIEVANCES_ADD_NOTE_AS_CREATOR',
  GRIEVANCES_ADD_NOTE_AS_OWNER: 'GRIEVANCES_ADD_NOTE_AS_OWNER',
  GRIEVANCES_SET_IN_PROGRESS: 'GRIEVANCES_SET_IN_PROGRESS',
  GRIEVANCES_SET_IN_PROGRESS_AS_CREATOR:
    'GRIEVANCES_SET_IN_PROGRESS_AS_CREATOR',
  GRIEVANCES_SET_IN_PROGRESS_AS_OWNER: 'GRIEVANCES_SET_IN_PROGRESS_AS_OWNER',
  GRIEVANCES_SET_ON_HOLD: 'GRIEVANCES_SET_ON_HOLD',
  GRIEVANCES_SET_ON_HOLD_AS_CREATOR: 'GRIEVANCES_SET_ON_HOLD_AS_CREATOR',
  GRIEVANCES_SET_ON_HOLD_AS_OWNER: 'GRIEVANCES_SET_ON_HOLD_AS_OWNER',
  GRIEVANCES_SEND_FOR_APPROVAL: 'GRIEVANCES_SEND_FOR_APPROVAL',
  GRIEVANCES_SEND_FOR_APPROVAL_AS_CREATOR:
    'GRIEVANCES_SEND_FOR_APPROVAL_AS_CREATOR',
  GRIEVANCES_SEND_FOR_APPROVAL_AS_OWNER:
    'GRIEVANCES_SEND_FOR_APPROVAL_AS_OWNER',
  GRIEVANCES_SEND_BACK: 'GRIEVANCES_SEND_BACK',
  GRIEVANCES_SEND_BACK_AS_CREATOR: 'GRIEVANCES_SEND_BACK_AS_CREATOR',
  GRIEVANCES_SEND_BACK_AS_OWNER: 'GRIEVANCES_SEND_BACK_AS_OWNER',
  GRIEVANCES_APPROVE_DATA_CHANGE: 'GRIEVANCES_APPROVE_DATA_CHANGE',
  GRIEVANCES_APPROVE_DATA_CHANGE_AS_CREATOR:
    'GRIEVANCES_APPROVE_DATA_CHANGE_AS_CREATOR',
  GRIEVANCES_APPROVE_DATA_CHANGE_AS_OWNER:
    'GRIEVANCES_APPROVE_DATA_CHANGE_AS_OWNER',
  GRIEVANCES_UNDO_APPROVED_DATA_CHANGE: 'GRIEVANCES_UNDO_APPROVED_DATA_CHANGE',
  GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_CREATOR:
    'GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_CREATOR',
  GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_OWNER:
    'GRIEVANCES_UNDO_APPROVED_DATA_CHANGE_AS_OWNER',
  GRIEVANCES_APPROVE_FLAG_AND_DEDUPE: 'GRIEVANCES_APPROVE_FLAG_AND_DEDUPE',
  GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_CREATOR:
    'GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_CREATOR',
  GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_OWNER:
    'GRIEVANCES_APPROVE_FLAG_AND_DEDUPE_AS_OWNER',
  GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK:
    'GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK',
  GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_CREATOR:
    'GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_CREATOR',
  GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_OWNER:
    'GRIEVANCES_CLOSE_TICKET_EXCLUDING_FEEDBACK_AS_OWNER',
  GRIEVANCES_CLOSE_TICKET_FEEDBACK: 'GRIEVANCES_CLOSE_TICKET_FEEDBACK',
  GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_CREATOR:
    'GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_CREATOR',
  GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_OWNER:
    'GRIEVANCES_CLOSE_TICKET_FEEDBACK_AS_OWNER',
};

export function hasPermissions(
  permission: string | string[],
  allowedPermissions: string[],
): boolean {
  // checks to see if has one permission or at least one from the array

  if (Array.isArray(permission)) {
    return allowedPermissions.some((perm) => permission.includes(perm));
  }
  return allowedPermissions.includes(permission);
}

export function hasPermissionInModule(
  module: string,
  allowedPermissions: string[],
): boolean {
  return allowedPermissions.some((perm) => perm.includes(module));
}
