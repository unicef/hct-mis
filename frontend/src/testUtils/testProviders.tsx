import MomentUtils from '@date-io/moment';
import { ThemeProvider } from '@material-ui/core';
import CssBaseline from '@material-ui/core/CssBaseline';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import moment from 'moment';
import React from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import { theme } from '../theme';

export const TestProviders: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>
    <StyledThemeProvider theme={theme}>
      <MuiPickersUtilsProvider libInstance={moment} utils={MomentUtils}>
        <CssBaseline />
        {children}
      </MuiPickersUtilsProvider>
    </StyledThemeProvider>
  </ThemeProvider>
);
