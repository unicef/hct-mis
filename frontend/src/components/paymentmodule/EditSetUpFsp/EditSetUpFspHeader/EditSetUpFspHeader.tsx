import React from 'react';
import { useTranslation } from 'react-i18next';
import { useParams } from 'react-router-dom';
import { hasPermissions, PERMISSIONS } from '../../../../config/permissions';
import { BreadCrumbsItem } from '../../../core/BreadCrumbs';
import { PageHeader } from '../../../core/PageHeader';

interface EditFspHeaderProps {
  businessArea: string;
  permissions: string[];
}

export const EditSetUpFspHeader = ({
  businessArea,
  permissions,
}: EditFspHeaderProps): React.ReactElement => {
  const { t } = useTranslation();
  const { id } = useParams();

  const breadCrumbsItems: BreadCrumbsItem[] = [
    {
      title: t('Payment Module'),
      to: `/${businessArea}/payment-module/payment-plans/${id}`,
    },
  ];

  return (
    <PageHeader
      title={t('Edit Set up FSP')}
      breadCrumbs={
        hasPermissions(
          PERMISSIONS.FINANCIAL_SERVICE_PROVIDER_UPDATE,
          permissions,
        )
          ? breadCrumbsItems
          : null
      }
    />
  );
};
