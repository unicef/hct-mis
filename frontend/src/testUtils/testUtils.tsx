import React, { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import cloneDeep from 'lodash/cloneDeep';
import { TestProviders } from './testProviders';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(<BrowserRouter>{ui}</BrowserRouter>, {
    wrapper: TestProviders,
    ...options,
  });

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export const makeApolloLoadingMock = (mocks) => {
  let mocksWithDelay = cloneDeep(mocks);
  for (const mock of mocksWithDelay) {
    mock.request.delay = 1e21;
  }
  return mocksWithDelay;
};
