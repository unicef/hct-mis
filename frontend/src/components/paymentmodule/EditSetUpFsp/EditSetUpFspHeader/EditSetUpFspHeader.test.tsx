import React from 'react';
import { PERMISSIONS } from '../../../../config/permissions';
import { render } from '../../../../testUtils/testUtils';
import { EditSetUpFspHeader } from './EditSetUpFspHeader';

describe('components/paymentmodule/EditSetUpFspHeader', () => {
  it('should render', () => {
    const { container } = render(
      <EditSetUpFspHeader
        handleSubmit={jest.fn()}
        businessArea='afghanistan'
        permissions={[PERMISSIONS.PAYMENT_MODULE_VIEW_LIST]}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
