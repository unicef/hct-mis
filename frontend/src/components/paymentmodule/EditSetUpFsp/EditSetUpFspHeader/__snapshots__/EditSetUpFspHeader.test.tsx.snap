// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/EditSetUpFsp/EditSetUpFspHeader should render 1`] = `
<div>
  <div
    class="sc-EHOje iaZfbD"
    data-cy="page-header-container"
  >
    <div
      class="sc-bZQynM ktisbs"
    >
      <div
        class="sc-gzVnrw coeMyM"
      >
        <div>
          <div
            class="MuiBox-root MuiBox-root-1"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
              data-cy="page-header-title"
            >
              Edit Set up FSP
            </h5>
            <div
              class="MuiBox-root MuiBox-root-2"
            />
          </div>
        </div>
        <div
          class="sc-htoDjs jWUwCq"
        />
      </div>
    </div>
    <div
      class="sc-iwsKbI fVvfjI"
    />
  </div>
</div>
`;
